### ISSUE TITLE
[Bug]: ad_astra:oxygen_loader crashing the server when the input is set to water from a PipeZ pipe

### ISSUE THREAD

**Issue by JohnsonKyer** [terrarium-earth/Ad-Astra#508]
### Bug Description

The server crashed & isn't able to reboot since I put a PipeZ pipe connected to a sink as "input" of water to craft Oxygen. 

### How to Reproduce?

1. Put Oxygen Loader
2. Put PipeZ pipe on a side. 
3. Put water in the pipe (here, from a sink).
4. World crashes and can't get loaded anymore. 

### Version

ad_astra-forge-1.20.1-1.15.17

### Mod Loader Version

Forge 47.2.20

### Mod Loader

Forge

### Logs or additional context

[17:47:36] [Server thread/ERROR]: Encountered an unexpected exception
net.minecraft.ReportedException: Ticking block entity
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:897) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:814) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at net.minecraft.client.server.IntegratedServer.m_5705_(IntegratedServer.java:89) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[?:?]
	at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[?:?]
	at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266) ~[?:?]
	at java.util.Objects.checkIndex(Objects.java:361) ~[?:?]
	at java.util.ArrayList.get(ArrayList.java:427) ~[?:?]
	at earth.terrarium.adastra.common.utils.TransferUtils.pullFluidNearby(TransferUtils.java:89) ~[ad_astra-forge-1.20.1-1.15.17.jar%23576!/:?]
	at earth.terrarium.adastra.common.blockentities.machines.OxygenLoaderBlockEntity.tickSideInteractions(OxygenLoaderBlockEntity.java:88) ~[ad_astra-forge-1.20.1-1.15.17.jar%23576!/:?]
	at earth.terrarium.adastra.common.blockentities.base.ContainerMachineBlockEntity.internalServerTick(ContainerMachineBlockEntity.java:51) ~[ad_astra-forge-1.20.1-1.15.17.jar%23576!/:?]
	at earth.terrarium.adastra.common.blockentities.base.EnergyContainerMachineBlockEntity.internalServerTick(EnergyContainerMachineBlockEntity.java:26) ~[ad_astra-forge-1.20.1-1.15.17.jar%23576!/:?]
	at earth.terrarium.adastra.common.blockentities.base.RecipeMachineBlockEntity.internalServerTick(RecipeMachineBlockEntity.java:31) ~[ad_astra-forge-1.20.1-1.15.17.jar%23576!/:?]
	at earth.terrarium.adastra.common.blocks.base.BasicEntityBlock.lambda$getTicker$2(BasicEntityBlock.java:65) ~[ad_astra-forge-1.20.1-1.15.17.jar%23576!/:?]
	at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.m_142224_(LevelChunk.java:689) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.m_142224_(LevelChunk.java:782) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at net.minecraft.world.level.Level.redirect$cpg000$observable$redirectTick(Level.java:4040) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at net.minecraft.world.level.Level.m_46463_(Level.java:468) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:351) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:893) ~[client-1.20.1-20230612.114412-srg.jar%23986!/:?]
	... 5 more
[17:47:36] [Server thread/FATAL]: Preparing crash report with UUID 9d56e149-2ef0-478c-be25-76357c75f07e
[17:47:36] [Server thread/ERROR]: This crash report has been saved to: C:\Users\Cyan\curseforge\minecraft\Instances\All the Mods 9 - ATM9\crash-reports\crash-2024-03-06_17.47.36-server.txt
[17:47:36] [Server thread/INFO]: Stopping server
[17:47:36] [Server thread/INFO]: Saving players
[17:47:36] [Server thread/INFO]: Saving worlds
[17:47:36] [Render thread/FATAL]: Preparing crash report with UUID a9f2c65d-f795-4a64-87a2-aac4fbf5f970
[17:47:41] [Server thread/INFO]: Loading JourneyMap Forge Configs

https://mclo.gs/ZgjwdMx

### Code of Conduct

- [X] I have searched the issue tracker and confirmed that the issue has not been reported
- [X] I have checked the FAQ (if one is present) and my issue is not listed
- [X] I have verified that I am using the latest version of the mod
- [X] I have verified that I have installed all the required dependencies for the mod
- [X] I have verified that I do not have Optifine installed. We will close issues if we detect Optifine is in your pack. We cannot fix Optifine issues. Please do not report issues with Optifine present.

**Comment by LetsBeChillx** [comment]
I also hit this on my All the Mods 9 server just a few minutes ago by connecting a Universal Pipe to an Oxygen Loader. The server crashed immediately when changing the pipe to Pull (shift + right-click pipe with wrench). I believe that face on the Oxygen Loader was set to Pull into the Input Fluid field, which was not pulling water from a Basic Fluid Tank through the pipe.

I can provide a crash log and versions if needed.

**Comment by CodexAdrian** [comment]
> I also hit this on my All the Mods 9 server just a few minutes ago by connecting a Universal Pipe to an Oxygen Loader. The server crashed immediately when changing the pipe to Pull (shift + right-click pipe with wrench). I believe that face on the Oxygen Loader was set to Pull into the Input Fluid field, which was not pulling water from a Basic Fluid Tank through the pipe.
> 
> I can provide a crash log and versions if needed.

Yes, please provide a crash log, ideally as a link. https://mclo.gs/

**Comment by LetsBeChillx** [comment]
>
> Yes, please provide a crash log, ideally as a link. https://mclo.gs/

Here you go! https://mclo.gs/vCI7OYV

I also noticed a second crash log, though I'm not sure if it's related or occurred when rebooting the server: https://mclo.gs/10MVc9s

EDIT: I was able to recreate this on a clean install with just Ad Astra and Pipez installed. With one of the Oxygen Loader's side set to Pull into the Input Fluid column, placing a fluid (or universal) pipe and right-clicking it to set to Pull will crash the game and corrupt the world (the world will instantly crash again on reload). New crash log here: https://mclo.gs/BYQJ6hk
